MIMXRT1050_EVK: Add USB support

mmahadevan108 opened the pull request:
### Summary of changes <!-- Required -->
Add USB support for MXRT1050_EVK

#### Impact of changes <!-- Optional -->
NONE

#### Migration actions required <!-- Optional -->
NONE

### Documentation <!-- Required -->
NONE

### Pull request type <!-- Required -->

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

### Test results <!-- Required -->
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR

### Reviewers <!-- Optional -->
@maclobdell

mmahadevan108 committed:
MIMXRT1050_EVK: Add USB support

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

0xc0170 reviewed (changes requested):
Add SPDX to new files where its not already

0xc0170 reviewed (dismissed):
The latest approval is for the changes, not the license, still to be fixed.

mmahadevan108 committed:
MIMXRT1050_EVK: Fix to linker script for non-cached region

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
KW41Z: Avoid using TARGET_NXP macro

Having Freescale and NXP macro causes compile from both
TARGET_Freescale and TARGET_NXP HAL folders.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
